go-controller/pkg/ovn/pods.go

```diff
@@ -26,21 +27,50 @@ func (oc *DefaultNetworkController) syncPods(pods []interface{}) error {
 	//
 	// TBD: Before this succeeds, add Pod handler should not continue to allocate IPs for the new Pods.
 	expectedLogicalPorts := make(map[string]bool)
+	vms := make(map[ktypes.NamespacedName]bool)
 	for _, podInterface := range pods {
 		pod, ok := podInterface.(*kapi.Pod)
 		if !ok {
 			return fmt.Errorf("spurious object in syncPods: %v", podInterface)
 		}
 
-		if !oc.isPodScheduledinLocalZone(pod) {
+		switchName, err := oc.getExpectedSwitchName(pod)
+		if err != nil {
+			return fmt.Errorf("failed geting expected switch name at syncPods: %v", err)
+		}
+		if kubevirt.IsPodLiveMigratable(pod) {
+			isStale, err := kubevirt.IsMigratedSourcePodStale(oc.watchFactory, pod)
+			if err != nil {
+				return err
+			}
+			// The stale pods are "old" information from a live migration
+			// so it's not needed to be in sync
+			if isStale {
+				continue
+			}
+			vm := kubevirt.ExtractVMNameFromPod(pod)
+			if vm != nil {
+				vms[*vm] = oc.isPodScheduledinLocalZone(pod)
+			}
+			annotation, err := util.UnmarshalPodAnnotation(pod.Annotations, ovntypes.DefaultNetworkName)
+			if err != nil {
+				continue
+			}
+			zoneContainsPodSubnet := false
+			switchName, zoneContainsPodSubnet = kubevirt.ZoneContainsPodSubnet(oc.lsManager, annotation)
+			// Don't allocate ip if this zone do not own the ip
+			if !zoneContainsPodSubnet {
+				continue
+			}
```

Comment:
Could we encapsulate all this in a method called as
```
err := kubevirt.SyncLiveMigratablePod(pod, oc.lsManager, oc.allocatePodIPsOnSwitch)
```

I think you will find out it shares a lot of code with the `kubevirt.AllocateSyncMigratablePodIPs` I suggested above so they can internally reuse code.

In this method you could also do what you in `SyncVirtualMachines` for that specific pod, there is no need to do all of it at the end of syncPods, is it?

Reply:
I am not sure, this is not doing any sync, it just filter stuff and we have already a SyncVirtualMachines at the end.